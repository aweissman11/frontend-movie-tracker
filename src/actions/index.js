export const getUserLoggedIn = (id, name) => {
  return {
    type: 'GET_USER_LOGGED_IN',
    user: { id, name }
  }
}

export const logUserOut = () => {
  return {
    type:  'LOG_USER_OUT'
  }
}

export const displayLogin = () => {
  return {
    type: 'DISPLAY_LOGIN'
  }
}

export const displaySignUp = () => {
  return {
    type: 'DISPLAY_SIGN_UP'
  }
}

export const setFavorites = (favorites = []) => {
  return {
    type: 'SET_FAVORITES',
    favorites
  }
}

export const updateFilters = (filters) => {
  return {
    type: 'UPDATE_FILTERS',
    filters
  }
}
export const updateSearchQuery = (searchQuery) => {
  return {
    type: 'UPDATE_QUERY',
    searchQuery
  }
}

export const setMovieList = (movies) => {
  return {
    type: 'SET_MOVIE_LIST',
    movies
  }
}

export const setHasErrored = (bool) => {
  return {
    type: 'SET_HAS_ERRORED',
    hasErrored: bool
  }
}

export const setIsOk = (bool) => {
  return {
    type: 'SET_IS_NOT_OK',
    notOk: bool
  }
}

export const isLoading = (bool) => {
  return {
    type: 'IS_LOADING',
    isLoading: bool
  }
}