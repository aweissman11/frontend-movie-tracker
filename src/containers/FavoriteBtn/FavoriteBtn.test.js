import React from 'react';
import { shallow, mount } from 'enzyme';

import { FavoriteBtn, mapDispatchToProps, mapStateToProps } from './index';
import * as Actions from '../../actions';


describe('FavoriteBtn', () => {
  let wrapper;
  let movies = [{
    movie_id: 1,
    user_id: 1,
    title: 'mock title',
    poster_path: 'mockpath',
    release_date: 20180101,
    vote_average: 5,
    overview: 'mock overview',
    genre_ids: [0, 1]
  }];
  let user = {
    id: 1
  }
  let favorites = [{
    movie_id: 1,
    user_id: 1,
    title: 'mock title',
    poster_path: 'mockpath',
    release_date: 20180101,
    vote_average: 5,
    overview: 'mock overview',
    genre_ids: [0, 1]
  }];

  beforeEach(() => {
    wrapper = shallow(<FavoriteBtn 
      user={user}
      movies={movies}
      favorites={favorites}
    />)
  })
  
  
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
  
  
  it('should toggle favorite on click', () => {
    const mockToggleFavorite = jest.fn();

    wrapper.instance().toggleFavorite = mockToggleFavorite;

    wrapper.find('.favorite-btn').simulate('click');

    expect(mockToggleFavorite).toHaveBeenCalled();
  });

  describe('callAddFavorite', () => {
    beforeEach(() => {

    });

    it('should call fetch with the correct parameters', async () => {
      const mockAddFavorite = jest.fn().mockImplementation(() => {
        return Promise.resolve({})
      });

      await wrapper.setState({
        userDataBaseFetch: {
          addFavorite: mockAddFavorite()
        }
      });

      await wrapper.instance().callAddFavorite(movies, user, favorites, 2);

      expect(mockAddFavorite).toHaveBeenCalledWith(movies)
    });
  });



  it('should be isFavorited if it has been favorited', () => {
    
  })

  describe.skip('mapStateToProps', () => {
    it('should map the state to props', () => {
      const mockUser = {
          name: 'Aaron',
          id: 1
      }

      const mockMovies = [
        {
          title: 'Die Hard',
          rating: 'R'
        },
        {
          title: 'Die Hard 2',
          rating: 'R'
        },
        {
          title: 'Die Hard with a Vengeance',
          rating: 'R'
        }
      ]

      const mockFavorites = [
        {
          title: 'Die Hard',
          rating: 'R'
        },
        {
          title: 'Die Hard 2',
          rating: 'R'
        },
        {
          title: 'Die Hard with a Vengeance',
          rating: 'R'
        }
      ]
      
      const expectedUser = mockUser
      const expectedMovies = mockMovies
      const expectedFavorites = mockFavorites
      const mockState = {
        user: {...mockUser},
        movies: mockMovies,
        favorites: mockFavorites
      }
      
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps.user).toEqual(expectedUser)
      expect(mappedProps.movies).toEqual(expectedMovies)
      expect(mappedProps.favorites).toEqual(expectedFavorites)
    })
    
  })
  
  describe.skip('mapDispatchToProps', () => {
    it('should map the dispatch to props', () => {
      const mockUser = {
        name: 'Aaron',
        id: 1
      }

      const mockFavorites = [
        {
          title: 'Die Hard',
          rating: 'R'
        },
        {
          title: 'Die Hard 2',
          rating: 'R'
        },
        {
          title: 'Die Hard with a Vengeance',
          rating: 'R'
        }
      ]

      const mockDispatch = jest.fn();
      const actionToDispatch = Actions.updateFavorites(mockFavorites)
      const mappedProps = mapDispatchToProps(mockDispatch);
      
      mappedProps.setFavorites(mockFavorites)

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    })
  })
})