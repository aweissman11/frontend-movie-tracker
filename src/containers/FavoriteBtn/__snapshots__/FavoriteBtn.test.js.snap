// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteBtn should match the snapshot 1`] = `undefined`;
